src/usersyncroniser.ts

```diff
@@ -372,4 +390,14 @@ export class UserSyncroniser {
     private displayNameForUser(discordUser): string {
         return `${discordUser.username}#${discordUser.discriminator}`;
     }
+
+    private async leave(intent: Intent, roomId: string, checkCache: boolean = true) {
+        const userId = intent.getClient().getUserId();
+        if (checkCache && ![null, "join", "invite"]
```

Comment:
We don't need to do this because a .leave should NO-OP quickly in synapse if the user isn't joined.

Reply:
Also tested this, and it doesn't do that, it keeps on sending leave events to synapse

Reply:
ugh